### PR TITLE
Add in more guidance about Docker images

### LINKED ISSUE #158: Create Docker image for development
Related to #157, while we are already going to be revamping each analysis and adding new ones, this is the best time to create a Docker image for this repo for development. 
This would help us manage dependencies have more consistent environments. Also in our efforts to make styler automatic, we can set it up in a Docker image as well. 

Some requirements we know of so far for this Docker image (this list is not at all complete - part of the reason we should build this Docker image as we go along): 
- R version 4.0 
- Tidyverse 
- devtools
- styler  - This should also allow us to automatically set `use_tidy_style()` before development so when it is called by the `render-notebooks.R` it will be styling consistently. 
- snakemake
- ComplexHeatmap

### PR DESCRIPTION
**Pull request by cansavvy**
## Purpose:

Closes #158 

Now that we have the docker image online https://github.com/AlexsLemonade/refinebio-examples/issues/158#issuecomment-673133192 I realized we need more words about how to manage versions of it as well as we need to add that to the PR checklist. 

I'm uncertain how practical/useful my current plan for managing the docker image versions is.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -70,14 +70,34 @@ The `render-notebooks.R` script adds a `bibliography:` specification in the `.Rm
...
+
+# Attach a tag to your docker image
+docker tag ccdl/refinebio-examples ccdl/refinebio-examples:dev
+
+# Push it!
+docker push ccdl/refinebio-examples:dev
+```
+After a few rounds of changes to the Dockerfile has occurred, we can make a new version tag.
+For example, if there is a `ccdl/refinebio-examples:v1` we can move to `v2`.
+
```
Comment (cbethell) [inline CONTRIBUTING.md:100]: We may want to add a note here on the command one would use to see the latest version tag as well. What do you think?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -7,11 +7,19 @@ What was your strategy for this new or edited analysis?
...
+
+### Formatting Checks
+* [ ] Removed any manual numbering of sections.
+* [ ] Removed any instances of chunk naming.
+* [ ] Spell checked any Rmd file or md file.
+* [ ] Comments and documentation are up to date.
+
+### Docker/Snakemake
+* [ ] Any not yet added packages needed for this analysis have been added to the Dockerfile and it successfully builds.
 * [ ] In the Docker container, run [snakemake for rendering](https://github.com/AlexsLemonade/refinebio-examples/CONTRIBUTING.md#how-to-re-render-the-notebooks)
```
Comment (cbethell) [inline .github/PULL_REQUEST_TEMPLATE.md:25]:
```suggestion
* [ ] In the Docker container, run [snakemake for rendering](https://github.com/AlexsLemonade/refinebio-examples/blob/master/CONTRIBUTING.md#how-to-re-render-the-notebooks)
```
This link was also broken

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -7,11 +7,19 @@ What was your strategy for this new or edited analysis?
 ## Concerns/Questions for reviewers:
 What things should reviewers look out for?
 
-## Pull Request Check List (rougly in order):
+## Analysis Pull Request Check List (roughly in order):
+
+### Content checks
+* [ ] All `{{BLANKS}}` have been replaced with the correct content.
 * [ ] [Sources are cited](https://github.com/AlexsLemonade/refinebio-examples/CONTRIBUTING.md#citing-sources-in-text)
```
Comment (cbethell) [inline .github/PULL_REQUEST_TEMPLATE.md:14]:
```suggestion
* [ ] [Sources are cited](https://github.com/AlexsLemonade/refinebio-examples/blob/master/CONTRIBUTING.md#citing-sources-in-text)
```

This link was broken

File: CONTRIBUTING.md (hunk 1 above)
Comment (cansavvy) [reply CONTRIBUTING.md:101]: It's the commands above but you just switch the tag. I added a sentence that says this. :+1:

**Pull request merged**